logstash-core/src/test/java/org/logstash/common/io/DeadLetterQueueReaderTest.java

```diff
@@ -134,9 +134,63 @@ public void testSeek() throws Exception {
         assertThat(entry.getReason(), equalTo("543"));
     }
 
+
     @Test
-    public void testInvalidDirectory()  throws Exception {
-        DeadLetterQueueReader reader = new DeadLetterQueueReader(dir);
-        assertThat(reader.pollEntry(100), is(nullValue()));
+    public void testWriteStopSmallWriteSeekByTimestamp() throws Exception {
+        DeadLetterQueueWriter writeManager = new DeadLetterQueueWriter(dir, 10000000, 10000000);
+        Event event = new Event(Collections.emptyMap());
+        Timestamp target = null;
+        long currentEpoch = System.currentTimeMillis();
+
+        for (int i = 0; i < 100; i++) {
+            DLQEntry entry = new DLQEntry(event, "foo", "bar", String.valueOf(i), new Timestamp(currentEpoch++));
+            writeManager.writeEntry(entry);
+            target = entry.getEntryTime();
+        }
+        writeManager.close();
+
+        writeManager = new DeadLetterQueueWriter(dir, 10000000, 10000000);
```

Comment:
@robbavey this thing is a `Closeable` now => we should be able to `try-with-resources` here :)